Firmware/Marlin_main.cpp

```diff
@@ -10236,13 +10242,14 @@ void bed_analysis(float x_dimension, float y_dimension, int x_points_num, int y_
 #endif //HEATBED_ANALYSIS
 
 #ifndef PINDA_THERMISTOR
-static void temp_compensation_start() {
-	
+static bool temp_compensation_start() {
+	bool retracted = false;
 	custom_message_type = CustomMsg::TempCompPreheat;
 	custom_message_state = PINDA_HEAT_T + 1;
 	lcd_update(2);
 	if ((int)degHotend(active_extruder) > extrude_min_temp) {
```

Comment:
If we want to keep this global, then maybe also check for  temp_compensation_retracted here, so that if MBL fails we don't do it again..